src/Faker/Provider/nl_NL/Person.php

```diff
@@ -346,4 +377,87 @@ public function idNumber()
         }
         return implode('', array_reverse($nr));
     }
+
+    /**
+     * @param string|null $documentCode Key from $idStartLetter
+     * @param array $date               'timestamp' of 'type' issue or expiry. 'birth' is birthdate of person
+     * @param string|null $timezone     Timezone to use
+     */
+    public function identityDocument(
+        $documentCode = null,
+        $date = [
+            'timestamp'  => null,
+            'type'       => 'issue',
+            'birth'      => null,
+        ],
+        $timezone = null
+    ) {
+        // Validity
+        if (isset($date['birth'])) {
+            $age = $date['birth']->diff(new \DateTime('today'))->y;
+        }
+
+        if (
+            isset($age)
+            && $age < 18
+        ) {
+            $validity = 5;
+        } else {
+            $validity = 10;
+        }
+
+        // Date info
+        if (
+            isset($date['type'])
+            && (
+                $date['type'] === 'expiry'
+            )
+        ) {
+            $dateType = 'expiry';
+        } else {
+            $dateType = 'issue';
+        }
+
+        $timeStamp = !isset($date['timestamp']) ? DateTime::dateTimeThisDecade('now', $timezone) : $date['timestamp'];
+        $ts = clone $timeStamp;
+        if ($dateType === 'expiry') {
+            $issueDate  = $ts->modify('-' . $validity . 'years');
+            $expiryDate = $timeStamp;
+        } else {
+            $issueDate  = $timeStamp;
+            $expiryDate = $ts->modify('+' . $validity . 'years');
+        }
+
+        // Document number
+        if (
+            !isset($documentCode)
+        ) {
+            $documentCode = static::randomKey(static::$idStartLetter);
+        } elseif (!array_key_exists($documentCode, static::$idStartLetter)) {
+            throw new \InvalidArgumentException($documentCode . ' is not a valid document code');
```

Comment:
Faker should not throw exceptions like this, it should generate valid fake data.